lib/error.js

```diff
@@ -78,6 +78,8 @@ var ERRNO = {
   SERVER_BUSY: 201,
   FEATURE_NOT_ENABLED: 202,
   BACKEND_SERVICE_FAILURE: 203,
+
+  INTERNAL_VALIDATION_ERROR: 998,
```

Comment:
It seemed useful to add an explicit errno for the equivalent of failing an internal assertion.